lib/Circuit.ts

```diff
@@ -6,7 +6,11 @@ import { isValidElement, type ReactElement } from "react"
 import { createInstanceFromReactElement } from "./fiber/create-instance-from-react-element"
 import { identity, type Matrix } from "transformation-matrix"
 
-type RootCircuitEventName = "asyncEffectComplete"
+type RootCircuitEventName =
+  | "asyncEffectComplete"
+  | "renderable:renderLifecycle:anyEvent"
+  | `renderable:renderLifecycle:${string}:start`
```

Comment:
Can we make this slightly better:

```suggestion
  | `renderable:renderLifecycle:${RenderPhase}:start`
```